sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/RelativePathPredicateTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SonarSource :: .NET :: Shared library
+ * Copyright (C) 2014-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.dotnet.tests;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.Collection;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.sonar.api.batch.fs.InputFile;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(Parameterized.class)
+public class RelativePathPredicateTest {
+
+  private String relativePath;
+  private String testInput;
+  private boolean expectedResult;
+
+  public RelativePathPredicateTest(String relativePath, String testInput, boolean expectedResult) {
+    this.relativePath = relativePath;
+    this.testInput = testInput;
+    this.expectedResult = expectedResult;
+  }
+
+  @Parameterized.Parameters
+  public static Collection input() {
+    return Arrays.asList(new Object[][] {
+        {"", "", true},
+        {"some/path/file.cs", "/_/some/path/file.cs", true},
+        {"some/path/file.cs", "/home/test/long/path/some/path/file.cs", true},
+        {"some/path/file.cs", "some/path/file.cs", true},
+        {"some/path/file.cs", "/path/file.cs", false},
+        {"SOME/path/file.cs", "some/path/file.cs", false},
+        {"foo", "bar", false},
+        {"foo", "", false},
+        {"\\some\\path\\file.cs", "/some/path/file.cs", false}
```

Comment:
Let's add this for completeness
```
 {"\\some\\path\\file.cs", "\\path\\file.cs", true}
```

Reply:
I had them initially, but the Java [URI class](https://docs.oracle.com/javase/7/docs/api/java/net/URI.html) implements the URI RFC specs which only use slashes - the constructor will throw when a backslash is in the string.